Mathlib/Data/ENNReal/Basic.lean

```diff
@@ -223,6 +223,9 @@ theorem ofReal_eq_coe_nnreal {x : ℝ} (h : 0 ≤ x) :
 
 @[simp] theorem toReal_nonneg {a : ℝ≥0∞} : 0 ≤ a.toReal := a.toNNReal.2
 
+@[simp] theorem toNNReal_toReal_eq (z : ℝ≥0∞) : z.toReal.toNNReal = z.toNNReal := by
+  ext; simp only [Real.coe_toNNReal', ge_iff_le, toReal_nonneg, max_eq_left]; rfl
```

Comment:
Why do you need the rfl here? Missing lemmas?

Reply:
The goal after this `simp only [...]`, before `rfl` was essentially the following lemma, which indeed seemed to be missing. However, creating it and making it a `@[simp]` lemma causes a loop in a simp later, so maybe there is a reason... On the other hand, the RHS below should definitely in my opinion be the simp normal form of this expression!

```
theorem coe_toNNReal_eq_toReal (z : ℝ≥0∞) : (z.toNNReal : ℝ) = z.toReal := rfl
```

I added it now without `@[simp]` tag, and providing this to `simp` now avoids the last `rfl`.